### PR TITLE
Update operators key management docs

### PR DESCRIPTION
**Pull request by samlaf**
As part of the recent efforts to upgrade our key management facilities (see https://docs.google.com/document/d/1qzXGca7A4S-9MnXhHluLGmGm7sYWvrln26nAH2Tc_Zc/edit#heading=h.2fblstx4oe2y), I was tasked with updating the docs for operator key management.

Main changes:
- renamed operator pages to instutional-operators and solo-operators for consistency: [f6dac7a](https://github.com/Layr-Labs/eigenlayer-docs/pull/218/commits/f6dac7a4d16a1f825ea0b5af09c816ccd9d5ad3c)
- added section for 1271 operators: [f210c7e](https://github.com/Layr-Labs/eigenlayer-docs/pull/218/commits/f210c7e2bc54d01e8d404ba03465e56b32a13c02) (needs a lot of love though..)
- added nomenclature + explanation regarding eigenlayer operator/avs keys: [cdfde05](https://github.com/Layr-Labs/eigenlayer-docs/pull/218/commits/cdfde0543fa4b58adcb501e05438d6368afb528e)

### CODE REVIEW

File: docs/eigenlayer/operator-guides/key-management/institutional-operators.md [hunk 1]
```diff
@@ -6,5 +6,10 @@ sidebar_position: 2
...
+- Given that Avs keys are likely to be much fewer, not using a seed to generate the keys is probably safer; generate each avs key independently if possible.
 - Use a remote signer like **[Web3signer](https://github.com/ConsenSys/web3signer)** or, better yet, distributed signers to eliminate single points of failure.
-- Develop a custom solution involving tailor-made tools. For instance, use Web3signer for remote signing and store keys on AWS Secrets Manager. A custom tool can manage automatic key storage in Secrets Manager and facilitate interactions with Web3signer.
\ No newline at end of file
+- Develop a custom solution involving tailor-made tools. For instance, use Web3signer for remote signing and store keys on AWS Secrets Manager. A custom tool can manage automatic key storage in Secrets Manager and facilitate interactions with Web3signer.
+
+# Smart Contract Operators
+
+We encourage institutional operators to register with eigenlayer using an [erc-1271](https://eips.ethereum.org/EIPS/eip-1271) smart contract wallet. This allows a lot more fine-grained control, such as multisig authorization and key rotation, which is currently not possible for EOA operators.
```
Comment (samlaf) [inline docs/eigenlayer/operator-guides/key-management/institutional-operators.md:15]: Added this as a starting template but someone with better knowledge should expand on this.

File: docs/eigenlayer/operator-guides/key-management/institutional-operators.md [hunk 2]
```diff
@@ -6,5 +6,10 @@ sidebar_position: 2
 
 - Secure keys, including secrets such as passphrases or mnemonics, using services like AWS Secrets Manager or Hashicorp Vault. These services can be seamlessly integrated with automated mechanisms that safely retrieve secrets or keys (e.g., remote signers). If resources permit, consider running your own Hashicorp Vault instance, which grants full custody of keys and secrets while sacrificing the service provider's availability and security guarantees.
 - Avoid generating all keys with the same mnemonic. Minimize the attack surface by employing a new mnemonic for every 200 or 1000 validator keys, depending on your preference. This approach also reduces the risk of losing key generation capabilities if a single mnemonic is lost, and limits the impact if an attacker gains access to a few mnemonics.
+- Given that Avs keys are likely to be much fewer, not using a seed to generate the keys is probably safer; generate each avs key independently if possible.
```
Comment (shrimalmadhur) [inline docs/eigenlayer/operator-guides/key-management/institutional-operators.md:9]: `not using a seed to generate the keys is probably safer` still not get why?

File: docs/eigenlayer/operator-guides/key-management/institutional-operators.md (hunk 1 above)
Comment (shrimalmadhur) [reply docs/eigenlayer/operator-guides/key-management/institutional-operators.md:15]: cc: @nelsonijih and @stevennevins

File: docs/eigenlayer/operator-guides/key-management/intro-key-management.md [hunk 3]
```diff
@@ -9,21 +9,38 @@ This section presents key management suggestions for operators. Within the conte
...
 
+### Eigenlayer
+
+Eigenlayer works differently from Ethereum due to it being natively a delegated PoS system. This means operators don't need to custody a separate key for each 32 Eth; they only need a single key per eigenlayer operator. This does imply that proper key management tools are even more important.
+
+Contrary to Eigenlayer, where an operator controls a bunch of (signing, withdrawal) key pairs, an eigenlayer operator controls a single operator key, as well as a separate key (or a small number of keys) for each AVS they are registered in.
+
+#### Operator keys
+
+An operator's operator key is the ecdsa key that is used to authenticate any interaction with the eigenlayer core contracts, and hence control an operator's actions such as registering to an operator, changing its operator parameters, force undelegating a staker, etc.
```
Comment (shrimalmadhur) [inline docs/eigenlayer/operator-guides/key-management/intro-key-management.md:38]: `force undelegating a staker` - do we have this now?

File: docs/eigenlayer/operator-guides/key-management/intro-key-management.md [hunk 4]
```diff
@@ -9,21 +9,38 @@ This section presents key management suggestions for operators. Within the conte
...
+
+The operator key should NEVER be used as an AVS key. They should also only be used to interact with the eigenlayer core contracts via the eigenlayer-cli or other operator-built tools, and should NOT be loaded into any node software. Any action that needs to be programatically triggered on the AVS contracts should be authorized via an AVS key, not the operator key.
+
+#### AVS signing keys
+
+AVS keys are actively used by node software to sign messages for AVSs. AVSs that use the eigenlayer-middleware contracts would typically use a bn254 key as the avs signing key, but other AVSs might require an ecdsa, ed25519, or any other key type.
```
Comment (shrimalmadhur) [inline docs/eigenlayer/operator-guides/key-management/intro-key-management.md:44]: `eigenlayer-middleware` - maybe add a link to this pointing to middleware contracts repo

File: docs/eigenlayer/operator-guides/key-management/intro-key-management.md [hunk 5]
```diff
@@ -9,21 +9,38 @@ This section presents key management suggestions for operators. Within the conte
...
+
+We advise operators to use a different key for every AVS, and NEVER to reuse their operator key as an AVS signing key.
```
Comment (shrimalmadhur) [inline docs/eigenlayer/operator-guides/key-management/intro-key-management.md:46]: I think there's a way to highlight it. like a `info` or `warn` block

we could also say. We strongly advise...  to be give very very strong advice

File: docs/eigenlayer/operator-guides/key-management/institutional-operators.md (hunk 2 above)
Comment (wesfloyd) [reply docs/eigenlayer/operator-guides/key-management/institutional-operators.md:9]: Can we capitalize all uses of "AVS" for consistency?

File: docs/eigenlayer/operator-guides/key-management/institutional-operators.md (hunk 1 above)
Comment (wesfloyd) [reply docs/eigenlayer/operator-guides/key-management/institutional-operators.md:15]: Would this include a Safe contract? Could we add any example smart contract wallets?

File: docs/eigenlayer/operator-guides/key-management/intro-key-management.md [hunk 6]
```diff
@@ -9,21 +9,38 @@ This section presents key management suggestions for operators. Within the conte
...
 
 The signing key, as explained above, is used for the validator to perform their duties. On the other hand, the *withdrawal key* has the power to control a validator's funds (transferring and withdrawing ETH).
 
 A validator should only need to use their withdrawal keys a few times over the lifetime of being a validator. This means they can be put into cold storage and stored with high security (offline).
 
+### Eigenlayer
+
+Eigenlayer works differently from Ethereum due to it being natively a delegated PoS system. This means operators don't need to custody a separate key for each 32 Eth; they only need a single key per eigenlayer operator. This does imply that proper key management tools are even more important.
+
+Contrary to Eigenlayer, where an operator controls a bunch of (signing, withdrawal) key pairs, an eigenlayer operator controls a single operator key, as well as a separate key (or a small number of keys) for each AVS they are registered in.
```
Comment (wesfloyd) [inline docs/eigenlayer/operator-guides/key-management/intro-key-management.md:34]: Did you mean "Contrary to Ethereum" here?

Comment (wesfloyd) [reply docs/eigenlayer/operator-guides/key-management/intro-key-management.md:34]: Can we capitalize all instances of "EigenLayer" for consistency sake?

File: docs/eigenlayer/operator-guides/key-management/intro-key-management.md (hunk 3 above)
Comment (wesfloyd) [inline docs/eigenlayer/operator-guides/key-management/intro-key-management.md:38]: Did you mean to say "registering to an AVS" here?

**Review by wesfloyd — COMMENTED** [review]
Added comments inline

File: docs/eigenlayer/operator-guides/key-management/institutional-operators.md (hunk 1 above)
Comment (stevennevins) [reply docs/eigenlayer/operator-guides/key-management/institutional-operators.md:15]: lgtm

Comment (samlaf) [reply docs/eigenlayer/operator-guides/key-management/institutional-operators.md:15]: @wesfloyd good question. 
@shrimalmadhur can we link to the example 1271 contracts you wrote for puffer or something?
@stevennevins do safe contracts abide to 1271 interface? If not how are they related to 1271 contracts? Are there any oeprators using safes right now? If so, can we add another section explaining this as well.

File: docs/eigenlayer/operator-guides/key-management/institutional-operators.md (hunk 2 above)
Comment (samlaf) [reply docs/eigenlayer/operator-guides/key-management/institutional-operators.md]: @wesfloyd [d2e4ea5](https://github.com/Layr-Labs/eigenlayer-docs/pull/218/commits/d2e4ea5a0fe82470fb1673b0292a870374dbee78)

Comment (samlaf) [reply docs/eigenlayer/operator-guides/key-management/institutional-operators.md]: @shrimalmadhur not sure if we should include that. Thinking is that if all your keys are generated from the same seed and you lose that seed they get all your keys. But if the keys are unrelated and they manage to steal one at least the others are safe.

File: docs/eigenlayer/operator-guides/key-management/intro-key-management.md (hunk 3 above)
Comment (samlaf) [reply docs/eigenlayer/operator-guides/key-management/intro-key-management.md:38]: yes, see https://github.com/Layr-Labs/eigenlayer-contracts/blob/dev/docs/core/DelegationManager.md#undelegate
Do you want me to link to this section as well?

File: docs/eigenlayer/operator-guides/key-management/intro-key-management.md (hunk 4 above)
Comment (samlaf) [reply docs/eigenlayer/operator-guides/key-management/intro-key-management.md]: [ea1b3a8](https://github.com/Layr-Labs/eigenlayer-docs/pull/218/commits/ea1b3a860a9418763f7cb4b438d16abfb3fef205)

File: docs/eigenlayer/operator-guides/key-management/intro-key-management.md [hunk 7]
```diff
@@ -9,21 +9,38 @@ This section presents key management suggestions for operators. Within the conte
...
+
+#### Operator keys
+
+An operator's operator key is the ecdsa key that is used to authenticate any interaction with the eigenlayer core contracts, and hence control an operator's actions such as registering to an operator, changing its operator parameters, force undelegating a staker, etc.
+
+The operator key should NEVER be used as an AVS key. They should also only be used to interact with the eigenlayer core contracts via the eigenlayer-cli or other operator-built tools, and should NOT be loaded into any node software. Any action that needs to be programatically triggered on the AVS contracts should be authorized via an AVS key, not the operator key.
```
[the remark this answers is not in the document]
Comment (samlaf) [reply docs/eigenlayer/operator-guides/key-management/intro-key-management.md]: [3840e97](https://github.com/Layr-Labs/eigenlayer-docs/pull/218/commits/3840e972c36f1331cdbd5b72378edb6aa4095d3a)

File: docs/eigenlayer/operator-guides/key-management/intro-key-management.md (hunk 5 above)
Comment (samlaf) [reply docs/eigenlayer/operator-guides/key-management/intro-key-management.md]: how about this? [b16685e](https://github.com/Layr-Labs/eigenlayer-docs/pull/218/commits/b16685e84116eb3c33b4f30f54345d496e279ec7)

File: docs/eigenlayer/operator-guides/key-management/intro-key-management.md (hunk 6 above)
Comment (samlaf) [reply docs/eigenlayer/operator-guides/key-management/intro-key-management.md]: definitely did. done: [3f0248c](https://github.com/Layr-Labs/eigenlayer-docs/pull/218/commits/3f0248c44a8f80360ebe86cd31f3a825faf03c38)

File: docs/eigenlayer/operator-guides/key-management/intro-key-management.md (hunk 3 above)
Comment (samlaf) [reply docs/eigenlayer/operator-guides/key-management/intro-key-management.md]: Yep, thanks for catching all the typos!! [b1bea0c](https://github.com/Layr-Labs/eigenlayer-docs/pull/218/commits/b1bea0c516b90c2bddfa98b2764d58c2448004de)

**Comment by samlaf** [comment]
@shrimalmadhur @wesfloyd reviewing all your comments. Please resolve those that you deem fixed.

**Review by dabit3 — DISMISSED** [review]

File: docs/eigenlayer/operator-guides/key-management/intro-key-management.md (hunk 3 above)
Comment (shrimalmadhur) [reply docs/eigenlayer/operator-guides/key-management/intro-key-management.md]: nah this is okay

File: docs/eigenlayer/operator-guides/key-management/institutional-operators.md (hunk 1 above)
Comment (shrimalmadhur) [reply docs/eigenlayer/operator-guides/key-management/institutional-operators.md]: I didn't write a contract. they wrote a contract which we could refer once they make it public - I just wrote a helper function to get eigenda registration parameters

File: docs/eigenlayer/operator-guides/key-management/institutional-operators.md (hunk 2 above)
Comment (shrimalmadhur) [reply docs/eigenlayer/operator-guides/key-management/institutional-operators.md]: `not using a same seed to` ? add `same` ?

File: docs/eigenlayer/operator-guides/key-management/institutional-operators.md [hunk 8]
```diff
@@ -6,5 +6,10 @@ sidebar_position: 2
 
 - Secure keys, including secrets such as passphrases or mnemonics, using services like AWS Secrets Manager or Hashicorp Vault. These services can be seamlessly integrated with automated mechanisms that safely retrieve secrets or keys (e.g., remote signers). If resources permit, consider running your own Hashicorp Vault instance, which grants full custody of keys and secrets while sacrificing the service provider's availability and security guarantees.
 - Avoid generating all keys with the same mnemonic. Minimize the attack surface by employing a new mnemonic for every 200 or 1000 validator keys, depending on your preference. This approach also reduces the risk of losing key generation capabilities if a single mnemonic is lost, and limits the impact if an attacker gains access to a few mnemonics.
+- Given that AVS keys are likely to be much fewer, not using a seed to generate the keys is probably safer; generate each AVS key independently if possible.
```
Comment (samlaf) [inline docs/eigenlayer/operator-guides/key-management/institutional-operators.md:9]:
```suggestion
- Given that AVS keys are likely to be much fewer, not using the same seed to generate the keys is probably safer; generate each AVS key independently if possible.
```

File: docs/eigenlayer/operator-guides/key-management/institutional-operators.md (hunk 2 above)
Comment (samlaf) [reply docs/eigenlayer/operator-guides/key-management/institutional-operators.md]: [de4d6cb](https://github.com/Layr-Labs/eigenlayer-docs/pull/218/commits/de4d6cbf68803c22cfc05e26c323a6b9dd3f408f)

**Review by shrimalmadhur — APPROVED** [review]

**Review by wesfloyd — APPROVED** [review]
lgtm

**Pull request merged**